packages/gatsby/src/utils/node-manifest.ts

```diff
@@ -60,21 +60,48 @@ const NODE_MANIFEST_FILE_LIMIT = getNodeManifestFileLimit()
  */
 async function findPageOwnedByNode({
   nodeId,
+  fullNode,
   slug,
 }: {
   nodeId: string
+  fullNode: IGatsbyNode
   slug?: string
 }): Promise<{
   page: INodeManifestPage
   foundPageBy: FoundPageBy
 }> {
   const state = store.getState()
   const { pages, nodes } = state
-  const { byNode } = state.queries
-
-  // the default page path is the first page found in
-  // node id to page query tracking
-  let pagePath = byNode?.get(nodeId)?.values()?.next()?.value
+  const { byNode, byConnection, trackedComponents } = state.queries
+
+  const { staticQueryComponents } = state
+  const nodeType = fullNode?.internal?.type
+
+  let pagePath =
+    // the default page path is the first page found in
+    // node id to page query tracking
+    byNode?.get(nodeId)?.values()?.next()?.value ||
```

Comment:
I prefer to keep the comments since it's not obvious what `byNode` and `byConnection` are and the comments give an explanation. This code is touched infrequently so it doesn't need to be read quickly - it's probably better for someone to take some time and read through and understand before working on this code.

Reply:
I can make variable names more descriptive like you're suggesting though, that's definitely a good idea